main.cpp

```diff
@@ -290,111 +304,32 @@ struct Numeric
                     return *this; 
                 }
             }
-            else if ( y < std::numeric_limits<Type>::epsilon() )
+            else if ( y < std::numeric_limits<T>::epsilon() )
```

Comment:
Either static_cast `y` to `T` or change `T` to `OtherType` to clear the warning:
```
main.cpp:314:19: warning: implicit conversion increases floating-point precision:
      'const float' to 'double' [-Wdouble-promotion]
        else if ( y < std::numeric_limits<T>::epsilon() )
                  ^ ~
main.cpp:375:7: note: in instantiation of function template specialization
      'Numeric<double>::operator/=<float>' requested here
    d /= 2.f;
```